src/main/java/io/kestra/plugin/weaviate/BatchCreate.java

```diff
@@ -99,17 +90,19 @@ public BatchCreate.Output run(RunContext runContext) throws Exception {
         }
 
         if (objects instanceof String) {
-            uri = Optional.of(URI.create((String) objects));
-            String outputFileContent = IOUtils.toString(storageInterface.get(uri.get()), Charsets.UTF_8);
-            Map parameters = JacksonMapper.ofYaml().readValue(outputFileContent, Map.class);
-
-            WeaviateObject weaviateObject = WeaviateObject.builder()
-                .id(UUID.randomUUID().toString())
-                .className(runContext.render(className))
-                .properties(parameters)
-                .build();
-
-            weaviateObjects.add(weaviateObject);
+            URI uri = URI.create((String) objects);
+            String renderedClassName = runContext.render(className);
+            InputStream inputStream = runContext.uriToInputStream(uri);
+
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+                String content = CharStreams.toString(new InputStreamReader(inputStream));
+                List<Map<String, Object>> parameters = List.of(JacksonMapper.ofYaml().readValue(content, Map.class));
+
+                List<WeaviateObject> weaviateObjectList = parameters.stream()
+                    .map(map -> WeaviateObject.builder().id(UUID.randomUUID().toString()).className(renderedClassName).properties(map).build())
+                    .toList();
+                weaviateObjects.addAll(weaviateObjectList);
```

Comment:
```
Flowable.create(FileSerde.reader(inputStream, Map.class), BackpressureStrategy.BUFFER)
                    .map(map -> WeaviateObject.builder()
                        .id(UUID.randomUUID().toString())
                        .className(runContext.render(className))
                        .properties(map)
                        .build()
                    ).toList().blockingGet()
```

Reply:
I did try this method earlier. But it throws: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.LinkedHashMap`
Not quite sure, might be i'm doing something wrong